flow/Arena.h

```diff
@@ -129,7 +130,13 @@ struct scalar_traits<Arena> : std::true_type {
 };
 
 struct ArenaBlockRef {
-	ArenaBlock* next;
+	// Only one of (next, aligned4kBuffer) are valid at any one time, as they occupy the same space.
```

Comment:
If you want this to have size 16 bytes, I think you need to reorder to put the fields that only require 4-byte alignment together. I think it's 24 bytes right now

Reply:
Will do.